pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ClientCnx.java

```diff
@@ -312,13 +332,20 @@ protected boolean isHandshakeCompleted() {
 
     CompletableFuture<LookupDataResult> newLookup(ByteBuf request, long requestId) {
         CompletableFuture<LookupDataResult> future = new CompletableFuture<>();
-        pendingLookupRequests.put(requestId, future);
-        ctx.writeAndFlush(request).addListener(writeFuture -> {
-            if (!writeFuture.isSuccess()) {
-                log.warn("{} Failed to send request to broker: {}", ctx.channel(), writeFuture.cause().getMessage());
-                future.completeExceptionally(writeFuture.cause());
-            }
-        });
+
+        if (addPendingLookupRequests(requestId, future)) {
+            ctx.writeAndFlush(request).addListener(writeFuture -> {
+                if (!writeFuture.isSuccess()) {
+                    log.warn("{} Failed to send request {} to broker: {}", ctx.channel(), requestId,
+                            writeFuture.cause().getMessage());
+                    future.completeExceptionally(writeFuture.cause());
+                }
+            });
+        } else {
+            log.warn("{} Failed to add lookup-request into pending queue", requestId);
```

Comment:
I'd rather keep this at debug level, since it is part of the normal operations flow